Shopify/cli
# Store config-specific env variables in separate dotenv file

### WHY are these changes introduced?

Fixes https://github.com/Shopify/internal-cli-foundations/issues/669

### WHAT is this pull request doing?

When config-as-code is enabled, store the app key and the extensions ids in a config-specific dotenv file.
For example, when using `shopify.app.staging.toml`, the dotenv file is going to be `.env.staging`.
If the user is using the default `shopify.app.toml`, we'll keep using the default `.env`

### How to test your changes?

- `pnpm create-app --local --template=node --path ~/Desktop`
- `pnpm shopify app config link --path ~/Desktop/your-app` (call the file `shopify.app.staging.toml`)
- `pnpm shopify app generate extension --path ~/Desktop/your-app`
- `pnpm shopify app deploy --path ~/Desktop/your-app` (the generated `.env` file should be `.env.staging`)

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
